### ISSUE TITLE
Diagnostics quickly disappear in the editor

### ISSUE THREAD

**Issue by matthewmichihara** [ballerina-platform/lsp4intellij#96]
**Description:**
I've set up a sample IntelliJ plugin that depends on the `lsp4intellij` library. I've configured it to talk to the lsp-sample server at https://github.com/microsoft/vscode-extension-samples/tree/master/lsp-sample. This language server returns diagnostics for words that are in all caps.

This is the result:
![diagnostics](https://user-images.githubusercontent.com/79073/59691170-9913e400-91b0-11e9-8d61-6b993293675c.gif)

Diagnostics briefly appear and then disappear in the IDE whenever I type. Stepping through with a debugger, I see that the right diagnostics are being returned by the language server, but for some reason they are quickly removed.

I've also tried building against https://github.com/ballerina-platform/lsp4intellij/pull/86 as @gayanper suggested in the gitter channel but started hitting exceptions due to what appears to be an unexpected null:
```
io.reactivex.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.IllegalArgumentException: Argument for @NotNull parameter 'file' of com/intellij/codeInsight/daemon/impl/DaemonCodeAnalyzerImpl.restart must not be null
    at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
    at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
    at io.reactivex.internal.observers.LambdaObserver.onError(LambdaObserver.java:77)
    at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:67)
    at io.reactivex.observers.SerializedObserver.onNext(SerializedObserver.java:111)
    at io.reactivex.internal.operators.observable.ObservableDebounceTimed$DebounceTimedObserver.emit(ObservableDebounceTimed.java:143)
    at io.reactivex.internal.operators.observable.ObservableDebounceTimed$DebounceEmitter.run(ObservableDebounceTimed.java:168)
    at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
    at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalArgumentException: Argument for @NotNull parameter 'file' of com/intellij/codeInsight/daemon/impl/DaemonCodeAnalyzerImpl.restart must not be null
    at com.intellij.codeInsight.daemon.impl.DaemonCodeAnalyzerImpl.$$$reportNull$$$0(DaemonCodeAnalyzerImpl.java)
    at com.intellij.codeInsight.daemon.impl.DaemonCodeAnalyzerImpl.restart(DaemonCodeAnalyzerImpl.java)
    at org.wso2.lsp4intellij.editor.EditorEventManager.lambda$null$0(EditorEventManager.java:245)
    at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:960)
    at org.wso2.lsp4intellij.utils.ApplicationUtils.computableReadAction(ApplicationUtils.java:49)
    at org.wso2.lsp4intellij.editor.EditorEventManager.lambda$new$1(EditorEventManager.java:242)
    at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
    ... 11 more
```

**Suggested Labels:**
Bug

**Suggested Assignees:**

**Affected Product Version:**
com.github.ballerina-platform:lsp4intellij:Tag:master-SNAPSHOT

**OS, DB, other environment details and versions:**    
OS: macOS Mojave
IntelliJ: 2019.1.2
JDK: 1.8

**Steps to reproduce:**
1. Clone and run https://github.com/matthewmichihara/sample-ballerina-plugin
2. With the newly running IJ instance, open the https://github.com/matthewmichihara/sample-ballerina-files project.
3. Open hello.txt in the editor.

Expected: The editor displays diagnostics for words that are written in all caps.
Actual: The editor briefly displays the diagnostics, but they quickly disappear.

**Comment by NipunaRanasinghe** [comment]
As per the gitter discussion, this should be fixed by @gayanper 's #86.. Lets get it merged as soon as after adding the required minor fixes to it.

**Comment by NipunaRanasinghe** [comment]
@matthewmichihara Shall we close this if this is fixed by #86? :)

**Comment by NipunaRanasinghe** [comment]
Closing since this should be fixed by #86 (credits should go to @gayanper for his amazing effort on this, as always!) 

@matthewmichihara please be kind enough to reopen/create an issue if you'll have any issues of the new implementation. :)
